### ISSUE TITLE
Login API endpoint

### ISSUE THREAD

**Issue by ChasNelson1990** [road86/bahis-oidc#1]
### Problem

I think this was discussed in September. Essentially, it's going to be easier to make v1 of the OAuth system be a little unorthodox. Right now this server expects us to forward the user to the server's login page and then passes the auth token back to the original app - however, because of the offline-first nature of bahis-desk this isn't ideal for us.

### Desired Solution

As we _do_ trust the server and client in this case, we want an API endpoint, e.g. `/login/`, that is a POST that accepts user credentials and returns and AuthToken.

**Comment by ChasNelson1990** [comment]
@shaarful - I actually thought you did this whilst we were in country? Can we make this a high priority please?

**Comment by shaarful** [comment]
@ChasNelson1990 Here is the endpoint for the dev server
authorization_base_url:  http://119.40.84.203:3000/authorize/
token_url: http://119.40.84.203:3000/token/
user_info_url: http://119.40.84.203:3000/userinfo/

**Comment by ChasNelson1990** [comment]
No, the authorize endpooint doesn't take a username and password as part of the request, right? It takes a client ID and callback URL and then provides it's own login form. Do you remember we discussed this and we want a new endpoint, e.g.
`curl --request POST --url http://119.40.84.203:3000/login/ --data '{"username":"XXX", "hashed_password":"XXX"}'` or something like this so that we can make a login request _without_ using the `/authorize/` form?
